source/solvers/tracer_scratch_data.cc

```diff
@@ -50,5 +54,39 @@ TracerScratchData<dim>::allocate()
 }
 
 
+template <int dim>
+void
+TracerScratchData<dim>::calculate_physical_properties()
+{
+  // Case where you have one fluid
+  if (properties_manager.get_number_of_fluids() == 1)
+    {
+      // In this case, only viscosity is the required property
+      const auto diffusivity_model =
+        properties_manager.get_tracer_diffusivity();
+      diffusivity_model->vector_value(fields, tracer_diffusivity);
+    }
+  else // properties_manager.get_number_of_fluids() == 2
+    {
+      // In this case,  we need both density and viscosity
+      const auto diffusivity_model_0 = properties_manager.get_density(0);
```

Comment:
Why did you choose to create two variables, rather than a vector containing them? (thus having `diffusivity_model[0]` or something like that)

Reply:
I don't have any good reason for that. I guess that was mostly laziness? I could make a routine for the property_manager that returns the vector of density model straight away and that could be a good way to work. I'll fix that.